tests/system/test_access.py

```diff
@@ -16,44 +20,316 @@
+        """
+        Test authorization logic for RUM Intake endpoint.
+        """
+        url = self.rum_intake_url
+        events = self.get_event_payload()
+
+        # Endpoint is not secured, all keys are expected to be allowed.
+        for token in self.authorized_keys + self.unauthorized_keys:
+            resp = requests.post(url, data=events, headers=headers(token))
+            assert resp.status_code != 401,  "token: {}, status_code: {}".format(token, resp.status_code)
+
+    def test_agent_config(self):
+        """
+        Test authorization logic for backend Agent Configuration endpoint
+        """
+        url = self.agent_config_url
+
+        for token in self.authorized_keys+[self.api_key_privilege_config]:
+            resp = requests.get(url, headers=headers(token, content_type="application/json"))
+            assert resp.status_code != 401,  "token: {}, status_code: {}".format(token, resp.status_code)
+
+        for token in self.unauthorized_keys+[self.api_key_privilege_intake, self.api_key_privilege_sourcemap]:
+            resp = requests.get(url, headers=headers(token, content_type="application/json"))
+            assert resp.status_code == 401,  "token: {}, status_code: {}".format(token, resp.status_code)
+
+    def test_rum_agent_config(self):
+        """
+        Test authorization logic for RUM Agent Configuration endpoint
+        """
+        url = self.rum_agent_config_url
+
+        # Endpoint is not secured, all keys are expected to be allowed.
+        for token in self.authorized_keys + self.unauthorized_keys:
+            resp = requests.get(url, headers=headers(token, content_type="application/json"))
+            assert resp.status_code != 401, "token: {}, status_code: {}".format(token, resp.status_code)
+
+    def test_sourcemap(self):
+        """
+        Test authorization logic for Sourcemap upload endpoint
+        """
+        def upload(token):
+            f = open(self._beat_path_join('testdata', 'sourcemap', 'bundle_no_mapping.js.map'))
+            resp = requests.post(self.sourcemap_url,
+                                 headers=headers(token, content_type="application/json"),
+                                 files={'sourcemap': f},
+                                 data={'service_version': '1.0.1',
+                                       'bundle_filepath': 'mapping.js.map',
+                                       'service_name': 'apm-agent-js'
+                                       })
+            return resp
+
+        for token in self.unauthorized_keys+[self.api_key_privilege_config, self.api_key_privilege_intake]:
+            resp = upload(token)
+            assert resp.status_code == 401, "token: {}, status_code: {}".format(token, resp.status_code)
+
+        for token in self.authorized_keys+[self.api_key_privilege_sourcemap]:
+            resp = upload(token)
+            assert resp.status_code != 401, "token: {}, status_code: {}".format(token, resp.status_code)
```

Comment:
These should 202?

Reply:
If RUM is enabled. But I can change to that.

Reply:
The `Authorization middleware` runs _before_ the `KillSwitchHandler middleware`. Without enabling `RUM` this would result in a `403` as it hits the kill switch _after_ the authorization. 

I can still enable the RUM endpoints and ensure test runs through.